### DISCUSSION TITLE
How to get test result status in teardown_method?

### DISCUSSION THREAD

**Discussion by joe-agent** [pytest-dev/pytest#9858] [Q&A]
```
class TestBase:
    def setup_method(self, method):
        print("setting up class")

    def teardown_method(self, method):
        print("tearing down class")
```

With a base class above, I have createed a class called `TestA` inheritnig from `TestBase` & I have my test method written in the `TestA` class like below:
```
class TestA(TestBase):
    def test_decorator(self): 
```

A log file will be created after each test method finishes running.
How do I delete the log file if the test has failed or skipped in the `teardown_method`?

Thanks

**Comment by portyu9** [comment]
Hi @joe-agent 

`teardown_method()` does not receive the test's `TestReport`.

A common supported pattern is to save the reports on the test item in `pytest_runtest_makereport`, then inspect them from a fixture after `yield`:

```python
import pytest


@pytest.hookimpl(wrapper=True)
def pytest_runtest_makereport(item, call):
    report = yield
    setattr(item, f"rep_{report.when}", report)


@pytest.fixture(autouse=True)
def cleanup_log(request):
    yield

    setup = getattr(request.node, "rep_setup", None)
    call = getattr(request.node, "rep_call", None)

    if (
        (setup is not None and setup.failed)
        or call is None
        or call.failed
        or call.skipped
    ):
        delete_log()
```

That lets the cleanup code distinguish a passed test from a failed/skipped one.

If you need to handle failures in the cleanup fixture's **own setup**, put the corresponding logic directly in pytest reporting hooks instead, because a fixture whose setup fails never reaches its `yield`.

Please mark this answer as accepted if it helped, thank you.
